docs/diagnostics/PX1019.md

```diff
@@ -0,0 +1,31 @@
+# PX1019
+This document describes the PX1019 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                                                                                | Type  | Code Fix    | 
+| ------ | ---------------------------------------------------------------------------------------------------------------- | ----- | ----------- | 
+| PX1019 | A DAC property field with the `AutoNumber` attribute must have the string type.                                  | Error | Available   |
+
+## Diagnostic Description
+You can use the `AutoNumber` attribute on a DAC field to enable auto-numbering for the field. The field must be of the string type.
```

Comment:
People usually ask about the reason for the diagnostics, so it's good to provide one in the docs. We can write something like:
"The AutoNumber attribute expects a field of string type. Fields of other types are not supported"